lib/active_shipping/carriers/canada_post_pws.rb

```diff
@@ -564,6 +578,36 @@ def log(msg)
 
     private
 
+    def contract?
+      @options[:contract_id].present?
+    end
+
+    SHIPMENT_NON_CONTRACT_MIMETYPE = "application/vnd.cpc.ncshipment+xml"
+    SHIPMENT_CONTRACT_MIMETYPE = "application/vnd.cpc.shipment-v7+xml"
+    def shipment_mimetype
+      contract? ? SHIPMENT_CONTRACT_MIMETYPE : SHIPMENT_NON_CONTRACT_MIMETYPE
+    end
+
+    RATE_NON_CONTRACT_MIMETYPE = "application/vnd.cpc.ship.rate+xml"
+    RATE_CONTRACT_MIMETYPE = "application/vnd.cpc.ship.rate-v3+xml"
+    def rate_mimetype
+      contract? ? RATE_CONTRACT_MIMETYPE : RATE_NON_CONTRACT_MIMETYPE
+    end
+
+    RATE_CONTRACT_URL = "http://www.canadapost.ca/ws/ship/rate-v3"
+    RATE_NON_CONTRACT_URL = "http://www.canadapost.ca/ws/ship/rate"
+    def rate_url
```

Comment:
I like this. Creating a variable just above the method, which describes what the difference is.